x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/OrdinalsGroupingOperator.java

```diff
@@ -512,16 +525,41 @@ public void close() {
 
     static final class BlockOrdinalsReader {
         private final SortedSetDocValues sortedSetDocValues;
+        private final SortedDocValues singleValues;
```

Comment:
Ideally we should have two classes for multi and single to avoid the if commands. I guess it requires more work, I am ok to keave it for a further optimisation. maybe add a comment?

Reply:
I do think it's more to make two classes.

Reply:
+1 I made two classes in https://github.com/elastic/elasticsearch/pull/108118/commits/734b6b53aa0e6c8c4984d44e84cb363fd561ebc2.